src/components/mortgage-investment-compare/index.js

```diff
@@ -22,100 +22,86 @@ import {
 
 const COMPOUND_FREQUENCY = 12
 
-const DEFAULT_OPTION = {
-  interestAmt: 0,
-  interestRate: 0,
-  term: 0,
-  pmt: 0,
-  fv: 0
-}
-
-let option1 = _.cloneDeep(DEFAULT_OPTION)
-let option2 = _.cloneDeep(DEFAULT_OPTION)
-
-option1.interestRate = 4
-option1.term = 15
-
-option2.interestRate = 4.3
-option2.term = 30
-
 class MortgageInvestmentCompare extends React.Component {
   constructor () {
     super()
-    this.toggleShowTable = this.toggleShowTable.bind(this)
-    this.calculatePMT = this.calculatePMT.bind(this)
-    this.calculateInterestAmt = this.calculateInterestAmt.bind(this)
     this.calculateFV = this.calculateFV.bind(this)
+    this.calculateInterestAmt = this.calculateInterestAmt.bind(this)
     this.calculateOpportunityCost = this.calculateOpportunityCost.bind(this)
+    this.calculatePMT = this.calculatePMT.bind(this)
+    this.getResult = this.getResult.bind(this)
+    this.toggleShowTable = this.toggleShowTable.bind(this)
     this.updateInput = this.updateInput.bind(this)
-    this.calculateAll = this.calculateAll.bind(this)
-    this.state = {
+
+    const input = {
+      inflation: 2,
       investmentRate: 9,
       loanAmt: 200000,
-      inflation: 2,
-      optCost: 0,
-      options: {
-        option1,
-        option2
+      option1: {
+        interestRate: 4,
+        term: 15
       },
-      shorterOption: '',
-      longerOption: '',
-      yearlyResultsByOption: [],
-      showTable: false
+      option2: {
+        interestRate: 4.3,
+        term: 30
+      }
     }
-  }
 
-  componentWillMount () {
-    // Running all calculations with default values on page load
-    this.setState(this.calculateAll(this.state))
+    const result = this.getResult(input)
```

Comment:
why not call this in a life cycle method like `componentWillMount`? It's a little strange to do this in React, unless there is some explicit reason for it.

Reply:
The initial state result doesn't depend on whether the component is mounted. This ties to your comment on setting the initial state for the `result`, see above. 

The difference is that we defined those values ourselves before. Now we ensure it's always the correct resulting calculation

Reply:
Ok. Undestood. You can ignore this then. I was under the impression that `getResult` was being called every time we changed the inputs in the actual state. I don't see anything particularly dangerous here but I will mention that I've worked with very experienced React devs and I've never seen them do something like this. They'll always define an initial state sometimes with empty values, and then populate it with lifecycle methods. I wonder if there's something we don't know...